File: TLS Inspector/View Controllers/HTTPHeadersTableViewController.m
#import "HTTPHeadersTableViewController.h"
#import "TitleValueTableViewCell.h"

@interface HTTPHeadersTableViewController ()

@property (strong, nonatomic) NSArray<NSString *> * headerKeys;

@end

@implementation HTTPHeadersTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.headerKeys = self.headers.allKeys;
}

- (BOOL) tableView:(UITableView *)tableView canPerformAction:(SEL)action forRowAtIndexPath:(NSIndexPath *)indexPath withSender:(id)sender {
    return action == @selector(copy:);
}

- (BOOL) tableView:(UITableView *)tableView shouldShowMenuForRowAtIndexPath:(NSIndexPath *)indexPath {
    return indexPath.section == 0;
}

- (void) tableView:(UITableView *)tableView performAction:(SEL)action forRowAtIndexPath:(NSIndexPath *)indexPath withSender:(id)sender {
    if (action == @selector(copy:)) {
        TitleValueTableViewCell * cell = [tableView cellForRowAtIndexPath:indexPath];
        [[UIPasteboard generalPasteboard] setString:cell.valueLabel.text];
    }
}

- (NSInteger) tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.headerKeys.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    NSString * key = self.headerKeys[indexPath.row];
    NSString * value = self.headers[key];

    TitleValueTableViewCell * cell = [[TitleValueTableViewCell alloc] initWithTitle:key value:value];
    [cell useFixedWidthFont];

    return cell;
}
@end

File: TLS Inspector/View Controllers/HTTPHeadersTableViewController.swift
import UIKit
import CertificateKit

class HTTPHeadersTableViewController: UITableViewController {

    var headers: [String:String] = [:]
    var headerKeysSorted: [String] = []

    override func viewDidLoad() {
        if let serverInfo = SERVER_INFO {
            self.headerKeysSorted = serverInfo.headers.keys.sorted()
            self.headers = serverInfo.headers
        }

        super.viewDidLoad()
    }

    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.headerKeysSorted.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let key = self.headerKeysSorted[indexPath.row]
        let value = self.headers[key] ?? ""

        return TitleValueTableViewCell.Cell(title: key, value: value, useFixedWidthFont: true)
    }

    override func tableView(_ tableView: UITableView, shouldShowMenuForRowAt indexPath: IndexPath) -> Bool {
        return true
    }

    override func tableView(_ tableView: UITableView, canPerformAction action: Selector, forRowAt indexPath: IndexPath, withSender sender: Any?) -> Bool {
        return action == #selector(copy(_:))
    }

    override func tableView(_ tableView: UITableView, performAction action: Selector, forRowAt indexPath: IndexPath, withSender sender: Any?) {
        if action == #selector(copy(_:)) {
            var data: String?
            let tableCell = tableView.cellForRow(at: indexPath)!
            if let titleValueCell = tableCell as? TitleValueTableViewCell {
                data = titleValueCell.valueLabel.text
            } else {
                data = tableCell.textLabel?.text
            }
            UIPasteboard.general.string = data
        }
    }
}
